### PR TITLE
Implementation of update/<status> API

### PR DESCRIPTION
**Pull request by kstribrnAmzn**
### Description of Changes
Implements a basic 'update/<status>' matcher which is used to verify the OTA job status update was accepted or rejected by AWS IoT Core.

### Manual testing steps
Built demo and executed an OTA download. The 'unhandled' message log is gone.

```
D Transport_OpenSSL: Successfully imported root CA.
D Transport_OpenSSL: Successfully imported client certificate.
D Transport_OpenSSL: Successfully imported client certificate private key.
D Transport_OpenSSL: Setting server name a77n9d666fj2p-ats.iot.us-east-1.amazonaws.com for SNI.
D Transport_OpenSSL: Established a TLS connection.
Successfully connected to IoT Core
Data topic is $aws/things/DevDesktopThing/streams/AFR_OTA-1702e4ff-26ce-4827-b3ba-63834b15f833/data/json
Get stream topic is $aws/things/DevDesktopThing/streams/AFR_OTA-1702e4ff-26ce-4827-b3ba-63834b15f833/get/json
Handled? 1SUBACK received with packet id: 2
MQTT streams handling incoming message 
Incoming Publish Topic Length: 90 Name: $aws/things/DevDesktopThing/streams/AFR_OTA-1702e4ff-26ce-4827-b3ba-63834b15f833/data/json matches subscribed topic.
Incoming Publish Message length: 43 Message: {"f":0,"i":0,"l":12,"p":"aGVsbG8gd29ybGQK"}
Downloaded Data hello world
 
OTA Completed successfully!
```

### Unit tests
100% line and branch coverage for new code

#### Is the change rollback friendly? Think about state in DynamoDB or other resources created by the new code and whether the old code can handle it. 
This introduces a new, unused API. No backwards compatibility concerns.

### CODE REVIEW

File: lib/iot-core-jobs/source/core_jobs.c [hunk 1]
```diff
@@ -55,37 +58,41 @@ static size_t getUpdateJobExecutionMsg( JobStatus_t status,
...
+bool coreJobs_isJobUpdateStatus( const char * topic,
+                                const size_t topicLength,
+                                const char * jobId,
+                                const size_t jobIdLength,
+                                JobUpdateStatus_t expectedStatus )
+{
+    /* Max suffix size = max topic size - "$aws/<thingname>" prefix */
+    char suffixBuffer[ TOPIC_BUFFER_SIZE - MAX_THING_NAME_LENGTH - 4U] = { 0 };
+    char jobIdTerminated[ MAX_JOB_ID_LENGTH + 1 ] = { 0 };
+    char updateStatusString[ 10 ] = { 0 };
```
Comment (manvensh) [inline lib/iot-core-jobs/source/core_jobs.c:82]: Can we change this magic number with a Macro?

File: lib/iot-core-jobs/source/core_jobs.c [hunk 2]
```diff
@@ -292,3 +299,38 @@ static size_t getUpdateJobExecutionMsg( JobStatus_t status,
...
+
+    if ( topic == NULL || topicLength == 0 )
+    {
+        isMatch = false;
+    }
+
+    if ( isMatch )
+    {
+        mqttWrapper_getThingName( thingName, &thingNameLength );
+        memcpy(&suffixTerminated, topicSuffix, topicSuffixLength);
```
Comment (manvensh) [inline lib/iot-core-jobs/source/core_jobs.c:323]: Nit: & can be removed here.

File: lib/iot-core-jobs/source/core_jobs.c (hunk 1 above)
Comment (kstribrnAmzn) [reply lib/iot-core-jobs/source/core_jobs.c:82]: Will do

File: lib/iot-core-jobs/source/core_jobs.c (hunk 2 above)
Comment (kstribrnAmzn) [reply lib/iot-core-jobs/source/core_jobs.c:323]: Will remove

**Review by manvensh — APPROVED** [review]

**Pull request merged**